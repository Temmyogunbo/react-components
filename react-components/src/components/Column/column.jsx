import React, { useEffect, useState } from "react";
import styled from "styled-components";
import { getClassValues } from "../../utils/getClassValues";
import { SCREEN_SIZE, SCREEN, MEDIA } from "../../constants";
import classNames from "classnames";

export const StyledColumn = styled.div`
  &.xs-Grid--full {
    flex: 0 0 100%;
  }

  &.xs-Grid--1of2 {
    flex: 0 0 50%;
  }

  &.xs-Grid--1of3 {
    flex: 0 0 33.3333%;
  }

  &.xs-Grid--1of4 {
    flex: 0 0 25%;
  }
  @media only screen and (min-width: ${MEDIA.small}px) {
    &.sm-Grid--full {
      flex: 0 0 100%;
    }
    &.sm-Grid--1of2 {
      flex: 0 0 50%;
    }
    &.sm-Grid--1of3 {
      flex: 0 0 33.3333%;
    }
    &.sm-Grid--1of4 {
      flex: 0 0 25%;
    }
  }

  @media only screen and (min-width: ${MEDIA.medium}px) {
    &.med-Grid--full {
      flex: 0 0 100%;
    }
    &.med-Grid--1of2 {
      flex: 0 0 50%;
    }
    &.med-Grid--1of3 {
      flex: 0 0 33.3333%;
    }
    &.med-Grid--1of4 {
      flex: 0 0 25%;
    }
  }

  @media only screen and (min-width: ${MEDIA.large}px) {
    &.lg-Grid--full {
      flex: 0 0 100%;
    }
    &.lg-Grid--1of2 {
      flex: 0 0 50%;
    }
    &.lg-Grid--1of3 {
      flex: 0 0 33.3333%;
    }
    &.lg-Grid--1of4 {
      flex: 0 0 25%;
    }
  }
`;
export const Column = React.forwardRef((props, ref) => {
  const [browserScreenSize, setBrowserScreenSize] = useState(SCREEN_SIZE.small);
  useEffect(() => {
    function onBrowserScreenChange() {
      const getScreensize = window.innerWidth;
      setBrowserScreenSize(getScreensize);
    }

    window.addEventListener("resize", () => onBrowserScreenChange());
    return window.removeEventListener("resize", () => onBrowserScreenChange());
  }, [browserScreenSize]);

  const classValues = getClassValues({ props, browserScreenSize });
  const columnClass = classNames(classValues);
  return (
    <StyledColumn {...props} className={columnClass} ref={ref}>
      {props.children}
    </StyledColumn>
  );
});
