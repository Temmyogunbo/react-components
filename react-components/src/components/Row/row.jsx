import React from "react";
import styled from "styled-components";

const StyledRow = styled.div`
& > * {
  flex: 1;
}
  display: flex;
  flex-wrap: wrap;
  margin: 0;
  padding: 0;
`;
export const Row = React.forwardRef((props, ref) => (<StyledRow className="row" ref={ref} {...props} >{props.children}</StyledRow>));
