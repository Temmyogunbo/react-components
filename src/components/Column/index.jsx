export { Column as default } from './column'
