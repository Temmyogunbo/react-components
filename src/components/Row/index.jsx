export { Row as default } from './row'