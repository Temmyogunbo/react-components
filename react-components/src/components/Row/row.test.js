import React from "react";
import Column from "../Column";
import Row from "../Row";
import { create } from "react-test-renderer";

describe("Row", () => {
  it("render child", () => {
    const testRenderer = create(
        <Row>
          <Column xs={1} sm={1} md={1} lg={1}>
            first child
          </Column>
          <Column xs={2} sm={2} md={2} lg={2}>
            seccond child
          </Column>
        </Row>
    );
    expect(testRenderer.toJSON()).toMatchSnapshot();
  });
});
