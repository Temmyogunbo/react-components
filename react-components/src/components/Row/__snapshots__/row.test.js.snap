// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Container render child 1`] = `
<div
  className="sc-bwzfXH eACkGl row"
>
  <div
    className="sc-bdVaJa bYFQGJ sm-Grid--full"
  >
    first child
  </div>
  <div
    className="sc-bdVaJa bYFQGJ sm-Grid--1of2"
  >
    seccond child
  </div>
</div>
`;

exports[`Row render child 1`] = `
<div
  className="sc-bwzfXH eACkGl row"
>
  <div
    className="sc-bdVaJa bYFQGJ sm-Grid--full"
  >
    first child
  </div>
  <div
    className="sc-bdVaJa bYFQGJ sm-Grid--1of2"
  >
    seccond child
  </div>
</div>
`;
